### PR TITLE
Sync: Defensive coding against E_NOTICE

### PR DESCRIPTION
**Pull request by kraftbj**
Fixes a series of E_NOTICEs thrown when testing on w.org:

```
E_NOTICE: Undefined index: pop in /home//public_html/wp-content/plugins/jetpack/vendor/automattic/jetpack-sync/src/class-rest-sender.php:47
```

```
E_NOTICE: Undefined index: encode in /home//public_html/wp-content/plugins/jetpack/vendor/automattic/jetpack-sync/src/class-rest-sender.php:66
```

```
E_NOTICE: Undefined index: encode in /home//public_html/wp-content/plugins/jetpack/vendor/automattic/jetpack-sync/src/class-rest-sender.php:73
```

#### Changes proposed in this Pull Request:

* Adds some checks.

#### Jetpack product discussion

n/a

#### Does this pull request change what data or activity we track or use?

No.

#### Testing instructions:

* Unclear how to reproduce, going off of log review.
*

### CODE REVIEW

File: projects/packages/sync/src/class-rest-sender.php [hunk 1]
```diff
@@ -62,15 +62,17 @@ public function queue_pull( $queue_name, $number_of_items, $args ) {
 			return new WP_Error( 'buffer_non-object', 'Buffer is not an object', 400 );
 		}
 
+		$encode = isset( $args['encode'] ) ? $args['encode'] : null;
```
Comment (mdbitz) [inline projects/packages/sync/src/class-rest-sender.php:65]: default state for encode needs to be true

**Review by mdbitz — CHANGES_REQUESTED** [review]
Thanks Kraft, can we have $encode default to true to match the default behavior of get_items_to_send?

**Review by mdbitz — APPROVED** [review]
Verified and approved.

**Review by anomiex — COMMENTED** [review]
Looks ok at a quick read-through, but needs a version bump. Also I see there's a use of `$args['force']` on line 51 that might want similar treatment.

An alternative option (feel free to ignore) to handle defaulting in `$args` would be to do something like this at the top of the method:
```php
// Supply defaults.
$args += array(
    'pop' => false,
    'force' => false,
    'encode' => true,
);
```
Maybe more clear (since we can't use `??`), at the cost of maybe doing unnecessary initialization.

**Comment by mdbitz** [comment]
Also note that I'm updating the Jetpack_Sync_Manager to send default values for these args: D66344

**Comment by sdixon194** [comment]
Cherry picked to release branch in 2ba49d0424272d0204f986a9924a31d8362888ac and released new version of Jetpack Sync

**Pull request merged**